lib/ansible/modules/network/aci/msc_label.py

```diff
@@ -0,0 +1,180 @@
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: msc_label
+short_description: Manage labels
+description:
+- Manage labels on Cisco ACI Multi-Site.
+author:
+- Dag Wieers (@dagwieers)
+version_added: '2.8'
+options:
+  label_id:
+    description:
+    - The ID of the label.
+    required: yes
+  label:
+    description:
+    - The name of the label.
+    required: yes
+    aliases: [ label_name, name ]
+  display_name:
+    description:
+    - The name of the label displayed in the web UI.
+  type:
+    description:
+    - The type of the label.
+    choices: [ site ]
+    default: site
+  state:
+    description:
+    - Use C(present) or C(absent) for adding or removing.
+    - Use C(query) for listing an object or multiple objects.
+    choices: [ absent, present, query ]
+    default: present
+extends_documentation_fragment: msc
+'''
+
+EXAMPLES = r'''
+- name: Add a new label
+  msc_label:
+    host: msc_host
+    username: admin
+    password: SomeSecretPassword
+    name: north_europe
+    label_id: 101
+    description: North European Datacenter
+    state: present
+  delegate_to: localhost
+
+- name: Remove a label
+  msc_label:
+    host: msc_host
+    username: admin
+    password: SomeSecretPassword
+    name: north_europe
+    state: absent
+  delegate_to: localhost
+
+- name: Query a label
+  msc_label:
+    host: msc_host
+    username: admin
+    password: SomeSecretPassword
+    name: north_europe
+    state: query
+  delegate_to: localhost
+  register: query_result
+
+- name: Query all labels
+  msc_label:
+    host: msc_host
+    username: admin
+    password: SomeSecretPassword
+    state: query
+  delegate_to: localhost
+  register: query_result
+'''
+
+RETURN = r'''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.network.aci.msc import MSCModule, msc_argument_spec, issubset
+
+
+def main():
+    argument_spec = msc_argument_spec()
+    argument_spec.update(
+        label=dict(type='str', required=False, aliases=['name', 'label_name']),
+        label_id=dict(type='str', required=False),
+        display_name=dict(type='str'),
+        type=dict(type='str', default='site', choices=['site']),
+        state=dict(type='str', default='present', choices=['absent', 'present', 'query']),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ['state', 'absent', ['label']],
+            ['state', 'present', ['label']],
+        ],
+    )
+
+    label = module.params['label']
+    label_id = module.params['label_id']
+    label_type = module.params['type']
+    state = module.params['state']
+
+    msc = MSCModule(module)
+
+    path = 'labels'
```

Comment:
Does it make sense to have the path be a property of the msc module utility?

Reply:
Yes, probably at some point. But for the more complex modules we will have to use a more complex interface, so this is working but temporary. I don't want to make an abstraction-layer for the easiest set of modules, if I know the more complex modules need a different abstraction layer.